Deploy Website with Github Pages (Related #776)

ClaasJG opened the pull request:
<!-- Before creating a PR, please make sure you read the contribution guidelines. -->
<!-- Feel free to delete points if they don't make sense for this PR. -->
<!-- You can tick boxes by putting an "x" inside the braces, or by clicking them once the comment is published. -->

<!-- Please use "Fixes #nr" and "Related #nr", respectively. -->
Hi, 

as described in #776, the domain is no longer available and therefore many of the documenting resources are no longer easily accessible.

This PR does not restore the domain,
but deploys the website and the book as well as the Rust API Docs with the help of Github Pages. 
It is roughly based on the conversation at [gitter](https://app.element.io/#/room/#slide-rs_specs:gitter.im/$Zcx-K8fr4wMBGAKtpWyGEgaBAQdmlSBPebTVBcSxnBI).

As long as this PR is open, https://claasjg.github.io/specs/ will reflect what the deploymend should look like.

It creates a new workflow, which first builds the book (tutorials) and the Rust API Doc as well as the website and then deploys them on GitHub Pages. This is independent of the CI workflow. (I.e. also independent of whether it fails. However, the deployment only takes place on the master branch).

In the repository under 'Settings' -> 'Pages' -> 'Build and deployment' -> 'Source' you have to switch from 'Deploy from a branch' to 'GitHub Actions' before the deployment step works.

This pull request contains a few more changes, which I can remove again or re-post in another PR. But I thought it would be a good idea to include them.

I have touched the website templates, because GitHub Pages are not delivered directly under the domain, but the repository name is placed in front of it and the base url was not respected correctly. 

Pages of the website that are no longer needed (references and some unused and empty about pages i.e.) as well as the old scripts to create the website with the help of netlify are removed.

In addition, some content from the readme has been added to the website so that visitors are no longer greeted by a blank page.

I hope that I have divided the individual changes into commits in such a way that individual changes can be easily adopted or discarded.

The link to the book in the readme has not yet been changed.

## Checklist

* [x] I've added tests for all code changes and additions (where applicable)
* [ ] I've added a demonstration of the new feature to one or more examples
* [x] I've updated the book to reflect my changes
* [ ] Usage of new public items is shown in the API docs

-ClaasJG

Imberflur commented on docs/website/content/_index.md:
```diff
@@ -1,5 +1,101 @@
+template = "section-nodate.html"
 +++
 
+> **S**pecs **P**arallel **ECS**
```
Thanks for making this look nicer!

Imberflur reviewed (commented):
Awesome stuff! Thanks for getting this together. I made a few suggestions.

Imberflur commented:
> In the repository under 'Settings' -> 'Pages' -> 'Build and deployment' -> 'Source' you have to switch from 'Deploy from a branch' to 'GitHub Actions' before the deployment step works.

Did this.

ClaasJG commented:
I think I have implemented all the suggestions (and also changed the link to the book to the future URL). Before the PR gets merged, I would squash the commits again like described and drop the one commit that slipped in by mistake (2034026).

Imberflur reviewed (approved):
Thanks!
